### PR TITLE
core[patch]: Add missing cache for create_model

### PR DESCRIPTION
**Pull request by eyurtsev**
It makes a big difference for performance.

### CODE REVIEW

File: libs/core/tests/unit_tests/language_models/chat_models/test_benchmark.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import time
+from itertools import cycle
+
+from langchain_core.language_models import GenericFakeChatModel
+
+
+def test_benchamrk_model() -> None:
```
Comment (eyurtsev) [inline libs/core/tests/unit_tests/language_models/chat_models/test_benchmark.py:7]:
```suggestion
def test_benchmark_model() -> None:
```

File: libs/core/langchain_core/runnables/utils.py [hunk 2]
```diff
@@ -789,9 +798,15 @@ def create_model(
...
         if isinstance(arg, tuple):
-            named_root_model = create_base_class(__model_name, arg[0], arg[1])
+            kwargs = {"type_": arg[0], "default_": arg[1]}
         else:
-            named_root_model = create_base_class(__model_name, arg)
+            kwargs = {"type_": arg}
+
+        try:
+            named_root_model = _create_root_model_cached(__model_name, **kwargs)
+        except TypeError:
```
Comment (baskaryan) [inline libs/core/langchain_core/runnables/utils.py:807]: ooc whats a TypeError here mean?

File: libs/core/tests/unit_tests/language_models/chat_models/test_benchmark.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        messages=cycle(["hello", "world", "!"]),
+    )
+
+    for _ in range(1_000):
+        model.invoke("foo")
+    toc = time.time()
+
+    # Verify that the time taken to run the loop is less than 0.5 seconds
+
+    assert (toc - tic) < 0.6
```
Comment (baskaryan) [inline libs/core/tests/unit_tests/language_models/chat_models/test_benchmark.py:21]: seems to take ~0.65 in ci :')

File: libs/core/langchain_core/runnables/utils.py (hunk 2 above)
Comment (eyurtsev) [reply libs/core/langchain_core/runnables/utils.py:807]: updated doc-string, but something wasn't hashable

**Pull request merged**